chunky/src/java/se/llbit/chunky/resources/Texture.java

```diff
@@ -1213,19 +1211,22 @@ public boolean isEmptyTexture() {
     return false;
   }
 
-  public Image fxImage() {
+  public javafx.scene.image.Image fxImage() {
     if (fxImage == null) {
-      fxImage = FxImageUtil.toFxImage(image);
+      fxImage = FxImageUtil.toFxImage(image.asBitmap());
     }
     return fxImage;
   }
 
-  /** Access the raw image data for this texture. */
-  public int[] getData() {
-    return image.data;
+  public BitmapImage getBitmap() {
+    return image.asBitmap();
   }
 
-  public BitmapImage getBitmap() {
-    return image;
+  /**
+   * Replace the current image by a compressed version of the image
+   */
+  public void compress() {
+    if(!(image instanceof CompressedImage))
+      image = new CompressedImage(image);
```

Comment:
When would we not want to compress a texture? Should this just be done in `setTexture`?

Reply:
Reading from a compressed texture is slower. When every texture access is slower, this has an impact on the rendering speed. I think it is undesirable to compress frequently used textures. When only rarely used texture are compressed, the slow down is not noticable